app/javascript/src/Common/ajaxEvents.ts

```diff
@@ -18,47 +18,31 @@ import $ from 'jquery'
 import type { AJAXErrorEvent } from 'Types/rails-ujs'
 
 /**
- * rails-ujs ajax events.
+ * rails-ujs ajax events. List of events: app/javascript/src/Types/rails-ujs.d.ts
  */
 $(document)
-  .on('ajax:before', (...args) => {
-    console.log('ajax:before on', args)
+  .on('ajax:before', () => {
     window.ThreeScale.showSpinner()
   })
-  .on('ajax:beforeSend', (...args) => { console.log('ajax:beforeSend', args) })
-  .on('ajax:send', (...args) => { console.log('ajax:send on', args) })
-  .on('ajax:stopped', (...args) => { console.log('ajax:stopped', args) })
-  .on('ajax:success', (...args) => { console.log('ajax:success', args) })
   .on('ajax:error', (event) => {
     const [response, status] = (event.originalEvent as AJAXErrorEvent<unknown>).detail
     console.error('ajax:error', response, status)
   })
-  .on('ajax:complete', (...args) => {
-    console.log('ajax:complete', args)
+  .on('ajax:complete', () => {
     window.ThreeScale.hideSpinner()
   })
 
 /**
- * window.$.ajax events, triggered by Colorbox.
- */
-window.$(document)
-  .on('ajaxStart', (...args) => { console.log('ajaxStart', args) })
-  .on('ajaxSend', (...args) => { console.log('ajaxSend', args) })
-  .on('ajaxSuccess', (...args) => { console.log('ajaxSuccess', args) })
-  .on('ajaxComplete', (...args) => { console.log('ajaxComplete', args) })
-  .on('ajaxError', (...args) => { console.log('ajaxError', args) })
-
-/**
- * Handle errors in formtastic forms rendered inside a colorbox. The error template is passed a
- * responseText inside the XHR response.
+ * Handle errors in formtastic forms that are rendered inside a colorbox. The error template is
+ * passed as responseText inside the XHR response.
  *
  * DEPRECATED: replace jquery/colorbox modals with Patternfly modals.
  * TODO: add a colorbox specific selector to separate this with non-legacy implementations.
  */
 $(document).on('ajax:error', 'form[data-remote]:not(.pf-c-form)', (event) => {
-  const [,, xhr] = (event.originalEvent as AJAXErrorEvent<HTMLDocument>).detail
+  const [,, xhr] = (event.originalEvent as AJAXErrorEvent<Document>).detail
 
-  // If EventTarget is replaceWith'ed immediately, ajax:success and ajax:complete are not called and the spinner will stay
+  // If EventTarget (modal's content) is replaced immediately, ajax:success and ajax:complete are not called and the spinner will stay
```

Comment:
So, does `setTimeout()` wait for all the events to be called and handled?...

Reply:
setTimeout 0 waits for the next cycle, which will happen after the current pile of events handlers are executed (due to JS being single thread).